### PR TITLE
Add edit topology timeouts

### LINKED ISSUE #169: Failed to edit topology: Timeout exceeded
I have a cluster of 100 instances (10 replicas with 10 instances). When deploying Cartridge role, I very often (4 times out of 5) get the next error in the "Edit Topology" step after 10-15 seconds of executing:
`fatal: [tnt_50_external_1_core_1_1 -> 10.0.0.102]: FAILED! => {"changed": false, "msg": "Failed to edit topology: Timeout exceeded"}`.

Let's make it possible to configure the maximum timeout.

### PR DESCRIPTION
**Pull request by hackallcode**
Closes #169

Before the patch, in a cluster of 100 instances, when deploying the cartridge role,
very often (4 times out of 5), you could receive the following error at the "Edit Topology" step
after 10-15 seconds of execution: "Failed to change topology: timed out".

Now it's possible to configure timeouts of two-phase commits on Cartridge 2.5+.

### CODE REVIEW

File: doc/scenario.md [hunk 1]
```diff
@@ -528,7 +528,11 @@ Input facts (set by config):
 - `failover_priority` - failover priority order;
 - `all_rw` - indicates that that all servers in the replicaset should be read-write;
 - `weight` - vshard replicaset weight;
-- `vshard_group` - vshard group.
+- `vshard_group` - vshard group;
+- `edit_topology_netbox_call_timeout` - time in seconds to wait netbox call while editing topology;
```
Comment (dokshina) [inline doc/scenario.md]: Note about minimum Cartridge version.

File: tasks/steps/edit_topology.yml [hunk 2]
```diff
@@ -12,6 +12,9 @@
         module_hostvars: '{{ cached_facts.edit_topology }}'
         play_hosts: '{{ play_hosts }}'
         console_sock: '{{ control_instance.console_sock }}'
-        timeout: '{{ edit_topology_timeout }}'
+        netbox_call_timeout: '{{ edit_topology_netbox_call_timeout }}'
```
Comment (dokshina) [inline tasks/steps/edit_topology.yml]: Maybe we can pass it as a one table variable? Like
```
opts:
  netbox_call_timeout: '{{ netbox_call_timeout }}'
  ..
```

File: doc/variables.md [hunk 3]
```diff
@@ -99,7 +99,13 @@ For more details see [scenario documentation](/doc/scenario.md).
 * `all_rw` (`boolean`): indicates that that all servers in the replicaset should be read-write;
 * `vshard_group` (`string`): name of vshard group the replicaset belongs to;
 * `weight` (`number`): vshard replicaset weight (matters only if `vshard-storage` role is enabled);
-* `edit_topology_timeout` (`number`, default: `60`): time in seconds to wait until a cluster become
+* `edit_topology_netbox_call_timeout` (`number`): time in seconds to wait netbox call
```
Comment (dokshina) [inline doc/variables.md]: Not sure that `edit_topology_` prefix is requred. Seems that `vars.options` affects everything, am I right?

File: module_utils/helpers.py [hunk 4]
```diff
@@ -415,6 +429,22 @@ def get_cluster_replicasets(control_console):
...
+
+
+def set_twophase_options_from_params(control_console, params):
+    twophase_options = {}
+    if params.get('netbox_call_timeout') is not None:
+        twophase_options['netbox_call_timeout'] = params['netbox_call_timeout']
+    if params.get('upload_config_timeout') is not None:
+        twophase_options['upload_config_timeout'] = params['upload_config_timeout']
+    if params.get('apply_config_timeout') is not None:
+        twophase_options['apply_config_timeout'] = params['apply_config_timeout']
```
Comment (olegrok) [inline module_utils/helpers.py]:
```py
allowed_options = [...]
twophase_options = {}
for option in allowed_options:
    if params.get(option) is not None:
        twophase_options[option] = params[option]
```

File: defaults/main.yml [hunk 5]
```diff
@@ -104,14 +104,20 @@ roles: null
 all_rw: null
 vshard_group: null
 weight: null
-edit_topology_timeout: 60
+
+twophase_netbox_call_timeout: null
+twophase_upload_config_timeout: null
+twophase_apply_config_timeout: null
+
+edit_topology_timeout: null  # DEPRECATED
```
Comment (olegrok) [inline defaults/main.yml:112]: I see no warnings `Option "edit_topology_timeout" is deprecated and will be removed in future release, use "edit_topology_healthy_timeout" instead`.

File: module_utils/helpers.py (hunk 4 above)
Comment (hackallcode) [reply module_utils/helpers.py]: The point is that specifically these options cannot be None, so we check them for None to filter.  Other options may be nullable, so they should not be checked.  Perhaps it is better to write that this is not_null_options

File: defaults/main.yml (hunk 5 above)
Comment (hackallcode) [reply defaults/main.yml:112]: It's a good idea to make warnings for deprecated variables!

Comment (hackallcode) [reply defaults/main.yml:112]: It's a good idea to make warnings for deprecated variables!

**Review by olegrok — DISMISSED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -21,6 +21,15 @@ README.md to use the newest tag with new release
 - `failover_promote` step to promote replicasets leaders
 - Allowed to skip user and group creation for tgz
 - Debug control instance and one not expelled instance
+- Timeouts for edit topology step:
```
Comment (dokshina) [inline CHANGELOG.md]: `edit_topology` and ... steps.

File: doc/scenario.md [hunk 7]
```diff
@@ -530,7 +530,15 @@ Input facts (set by config):
 - `failover_priority` - failover priority order;
 - `all_rw` - indicates that that all servers in the replicaset should be read-write;
 - `weight` - vshard replicaset weight;
-- `vshard_group` - vshard group.
+- `vshard_group` - vshard group;
+- `twophase_netbox_call_timeout` - time in seconds to wait netbox call
+  while editing topology (Cartridge 2.5+ is required);
```
Comment (dokshina) [inline doc/scenario.md]: editing topology -> changing cluster-wide config?

**Review by dokshina — APPROVED** [review]

**Pull request merged**